### PR TITLE
Update docs and examples for log4j2 usage.

### PR DESCRIPTION
**Pull request by housejester**
Druid no longer works with just log4j.xml as described in the docs (since it recently bumped up to using log4j2). Even if log4j.xml IS on the classpath (first even), it will be ignored, in favor of the log4j2.xml found in the druid jars. This should probably be spelled out as a difference for upgrading to Druid 0.7. 

This PR updates docs and examples to clarify log4j2.xml as well as clarify how the classpath should be set (ie the dir with the log4j2.xml should be before the jars). 

In this PR:
- Put configs early in classpath in examples so log4j2.xml will get picked up properly
- Add an example log4j2.xml file.
- Update Logging doc.

This PR does NOT address jdk logging. Any jdk logging does not respect the log4j configs. For example, if you set up logging to go only to file and not console, you will see jdk logging still go to console (mostly from jersey code). This can be addressed as well by:

- adding a dependency on log4j-jul to the parent and common poms:

```
<dependency>
	<groupId>org.apache.logging.log4j</groupId>
	<artifactId>log4j-jul</artifactId>
	<version>${log4j.version}</version>
</dependency>
```

- then add the following to the command line

```
-Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager
```

Alternatively, the system prop could be set in io.druid.cli.Main, but that didn't seem right. See http://logging.apache.org/log4j/2.0/log4j-jul/index.html for more info.

### CODE REVIEW

File: docs/content/Logging.md [hunk 1]
```diff
@@ -6,34 +6,35 @@ Logging
...
-<!DOCTYPE log4j:configuration SYSTEM "log4j.dtd">
-
-<log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
 
-  <appender name="ConsoleAppender" class="org.apache.log4j.ConsoleAppender">
-    <layout class="org.apache.log4j.PatternLayout">
-      <param name="ConversionPattern" value="%d{ISO8601} %-5p [%t] %c - %m%n"/>
-    </layout>
-  </appender>
+Note the "-classpath" in this example has the confid dir before the jars under lib/*.
```
Comment (nishantmonu51) [inline docs/content/Logging.md]: spelling correction - s/confid/config/g

File: docs/content/Logging.md [hunk 2]
```diff
@@ -6,34 +6,35 @@ Logging
...
+  <Appenders>
+    <Console name="Console" target="SYSTEM_OUT">
+      <PatternLayout pattern="%d{ISO8601} %p [%t] %c - %m%n"/>
+    </Console>
+  </Appenders>
+  <Loggers>
+    <Logger name="io.druid.curator.inventory.CuratorInventoryManager" level="warn"/>
+    <Logger name="io.druid.client.BatchServerInventoryView" level="warn"/>
+    <Logger name="io.druid.client.ServerInventoryView" level="warn"/>
+    <Logger name ="com.metamx.http.client.pool.ChannelResourceFactory" level="warn"/>
```
Comment (nishantmonu51) [inline docs/content/Logging.md]: can we remove setting explicit warn levels in default config file ?

File: examples/config/_common/log4j2.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <Appenders>
+        <Console name="Console" target="SYSTEM_OUT">
+            <PatternLayout pattern="%d{ISO8601} %p [%t] %c - %m%n"/>
+        </Console>
+    </Appenders>
+    <Loggers>
+        <Logger name="io.druid.curator.inventory.CuratorInventoryManager" level="warn"/>
+        <Logger name="io.druid.client.BatchServerInventoryView" level="warn"/>
+        <Logger name="io.druid.client.ServerInventoryView" level="warn"/>
+        <Logger name ="com.metamx.http.client.pool.ChannelResourceFactory" level="warn"/>
```
Comment (nishantmonu51) [inline examples/config/_common/log4j2.xml]: remove setting explicit log levels for these classes ?

File: docs/content/Logging.md (hunk 2 above)
Comment (housejester) [inline docs/content/Logging.md]: These were in the original doc as well as in the Performance FAQ, so it seemed they were likely candidates for tuning. Should I remove them (from here and the xml file) anyway?

**Comment by drcrallen** [comment]
log4j2 original PR: https://github.com/druid-io/druid/pull/1011
Just as an FYI

**Comment by drcrallen** [comment]
By and large Druid uses the [MMX logger](https://github.com/metamx/java-util/blob/master/src/main/java/com/metamx/common/logger/Logger.java).

Is jul for capturing non-druid logging? or is it included for some other reason?

File: docs/content/Logging.md (hunk 2 above)
Comment (drcrallen) [inline docs/content/Logging.md]: Performance also says [async logging can be enabled](https://github.com/druid-io/druid/blob/master/docs/content/Performance-FAQ.md#my-logs-are-really-chatty-can-i-set-them-to-asynchronously-write)

Comment (drcrallen) [inline docs/content/Logging.md]: In general a minimalist example is probably better (like https://github.com/druid-io/druid/blob/master/common/src/main/resources/log4j2.xml), or else people will think this is the recommended setup.

File: docs/content/Tutorial:-Loading-Batch-Data.md [hunk 4]
```diff
@@ -66,19 +66,19 @@ Note: If Zookeeper and MySQL aren't running, you'll have to start them again as
 To start the Indexing Service:
 
 ```bash
-java -Xmx2g -Duser.timezone=UTC -Dfile.encoding=UTF-8 -classpath lib/*:<hadoop_config_path>:config/_common:config/overlord io.druid.cli.Main server overlord
+java -Xmx2g -Duser.timezone=UTC -Dfile.encoding=UTF-8 -classpath config/_common:config/overlord:lib/*:<hadoop_config_path> io.druid.cli.Main server overlord
```
Comment (drcrallen) [inline docs/content/Tutorial:-Loading-Batch-Data.md]: Can this be `$(hadoop classpath)`?

File: examples/config/_common/log4j2.xml (hunk 3 above)
Comment (drcrallen) [inline examples/config/_common/log4j2.xml]: IMHO this should simply be a copy of https://github.com/druid-io/druid/blob/master/common/src/main/resources/log4j2.xml

File: docs/content/Logging.md (hunk 2 above)
Comment (xvrl) [inline docs/content/Logging.md]: We should probably update the docs in a separate PR to make the recommended logging the same as the default one, and indicate in the performance section the drawbacks of using async logging and why we don't recommend it by default.

**Comment by housejester** [comment]
@drcrallen Yes, the jul dependency would be for non-druid classes that druid winds up using. To be clear, the jul change is not included in this PR, I just thought it worth mentioning/discussing whether it should be (jcl too for that matter). 

If I'm managing a Druid service in production, and trying to tune the logging to only go to file or otherwise tune the levels (or have a consistent log format), Druid out of the box doesn't allow me to do that only using a log4j2.xml file. This is because some components (like jersey) are using jul, and that logger is not configured so it will just go to console. I'd either need to add a separate jul logging config, or drop in the log4j jul jar and set the command line arg myself. Same for jcl (which seems like is used by dbcp which is used in the db metadata connector stuff.

**Comment by drcrallen** [comment]
@housejester : Thanks for the jul info!

I just tried it out and here is an example of the change:

### No JUL
```
2015-02-19T15:54:03,259 INFO [main] org.eclipse.jetty.server.Server - jetty-9.2.5.v20141112
Feb 19, 2015 3:54:03 PM com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory register
INFO: Registering com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider as a provider class
Feb 19, 2015 3:54:03 PM com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory register
INFO: Registering io.druid.server.StatusResource as a root resource class
Feb 19, 2015 3:54:03 PM com.sun.jersey.server.impl.application.WebApplicationImpl _initiate
INFO: Initiating Jersey application, version 'Jersey: 1.17.1 02/28/2013 12:47 PM'
Feb 19, 2015 3:54:03 PM com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory getComponentProvider
INFO: Binding com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider to GuiceManagedComponentProvider with the scope "Singleton"
Feb 19, 2015 3:54:04 PM com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory getComponentProvider
INFO: Binding io.druid.server.QueryResource to GuiceInstantiatedComponentProvider
Feb 19, 2015 3:54:04 PM com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory getComponentProvider
INFO: Binding io.druid.server.http.HistoricalResource to GuiceInstantiatedComponentProvider
Feb 19, 2015 3:54:04 PM com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory getComponentProvider
INFO: Binding io.druid.server.StatusResource to GuiceManagedComponentProvider with the scope "Undefined"
2015-02-19T15:54:04,362 INFO [main] org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@331372c6{/,null,AVAILABLE}
2015-02-19T15:54:04,410 INFO [main] org.eclipse.jetty.server.ServerConnector - Started ServerConnector@e5322ec{HTTP/1.1}{0.0.0.0:8081}
```

### With JUL
```
2015-02-19T18:07:05,707 INFO [main] org.eclipse.jetty.server.Server - jetty-9.2.5.v20141112
2015-02-19T18:07:05,795 INFO [main-SendThread(localhost:2181)] org.apache.zookeeper.ClientCnxn - Socket connection established to localhost/127.0.0.1:2181, initiating session
2015-02-19T18:07:05,832 INFO [main-SendThread(localhost:2181)] org.apache.zookeeper.ClientCnxn - Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x14b98a1fbad0055, negotiated timeout = 30000
2015-02-19T18:07:05,838 INFO [main-EventThread] org.apache.curator.framework.state.ConnectionStateManager - State change: CONNECTED
2015-02-19T18:07:05,981 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Registering com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider as a provider class
2015-02-19T18:07:05,984 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Registering io.druid.server.StatusResource as a root resource class
2015-02-19T18:07:05,988 INFO [main] com.sun.jersey.server.impl.application.WebApplicationImpl - Initiating Jersey application, version 'Jersey: 1.17.1 02/28/2013 12:47 PM'
2015-02-19T18:07:06,116 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider to GuiceManagedComponentProvider with the scope "Singleton"
2015-02-19T18:07:06,874 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.QueryResource to GuiceInstantiatedComponentProvider
2015-02-19T18:07:06,889 INFO [ServerInventoryView-0] io.druid.client.BatchServerInventoryView - Inventory Initialized
2015-02-19T18:07:06,893 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.http.BrokerResource to GuiceInstantiatedComponentProvider
2015-02-19T18:07:06,896 INFO [ServerInventoryView-0] io.druid.curator.discovery.CuratorServiceAnnouncer - Announcing service[DruidNode{serviceName='broker', host='localhost', port=8080}]
2015-02-19T18:07:06,902 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.ClientInfoResource to GuiceInstantiatedComponentProvider
2015-02-19T18:07:06,903 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.StatusResource to GuiceManagedComponentProvider with the scope "Undefined"
2015-02-19T18:07:06,947 INFO [main] org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@6e21638f{/,null,AVAILABLE}
2015-02-19T18:07:06,965 INFO [main] org.eclipse.jetty.server.ServerConnector - Started ServerConnector@594dc4dc{HTTP/1.1}{0.0.0.0:8080}
```

Definitely something that would be nice as a separate PR (shouldn't need to block this one)

**Comment by drcrallen** [comment]
Also as a general FYI, here's where the original Log4j2 information was posted:
https://groups.google.com/forum/#!searchin/druid-development/log4j2.xml/druid-development/MVTAVAE9LUo/uoWHrolSC9cJ

**Comment by drcrallen** [comment]
Please squash then :+1:

**Pull request merged**